crates/transport-http/src/hyper_transport.rs

```diff
@@ -1,63 +1,78 @@
-use crate::{Http, HttpConnect};
 use alloy_json_rpc::{RequestPacket, ResponsePacket};
 use alloy_transport::{
     utils::guess_local_url, TransportConnect, TransportError, TransportErrorKind, TransportFut,
 };
 use http_body_util::{BodyExt, Full};
 use hyper::{
-    body::{Buf, Bytes},
-    header,
+    body::{Bytes, Incoming},
+    header, Request, Response,
 };
-use hyper_util::client::legacy::{connect::Connect, Client};
-use std::task;
+use hyper_util::client::legacy::Error;
+use std::{future::Future, marker::PhantomData, pin::Pin, task};
 use tower::Service;
 use tracing::{debug, debug_span, trace, Instrument};
+use url::Url;
+
+use crate::{Http, HttpConnect};
 
 /// A [`hyper`] HTTP client.
 pub type HyperClient = hyper_util::client::legacy::Client<
     hyper_util::client::legacy::connect::HttpConnector,
     http_body_util::Full<::hyper::body::Bytes>,
 >;
 
-/// An [`Http`] transport using [`hyper`].
-pub type HyperTransport = Http<HyperClient>;
+/// A [hyper] client that can be used with tower layers.
+#[derive(Clone, Debug)]
+pub struct HyperTransport<B = Full<Bytes>, S = HyperClient> {
```

Comment:
I am wondering if removing `Http` here is fine? I think it's possible to keep it, though unsure how useful is it in general

Reply:
I'd prefer us to keep `HyperTransport = Http<..>` type alias here, because right now `HyperTransport` on its own is not a transport

We basically only need the new type as a wrapper around hyper client with phatomdata, so we can rename it into `HyperClient`